src/rec_prithvi.py

```diff
@@ -23,16 +26,39 @@ def rec_run_prithvi(
         Job ID
     llm : str, optional
         LLM to be used, by default "claude-opus-4-20250514"
+    az_model : str, optional
+        AZ model to be used, by default "USPTO"
+    stability_flag : str, optional
+        Stability flag, by default "False"
+    hallucination_check : str, optional
+        Hallucination check, by default "False"
+    use_protecting_group_feature : bool, optional
+        Use protecting group feature, by default False
+    visited : set, optional
+        Set of molecules already processed (for cycle detection), by default None
+    depth : int, optional
+        Current recursion depth, by default 0
+    max_depth : int, optional
+        Maximum recursion depth, by default 50
 
     Returns
     -------
     tuple(dict, bool)
         result_dict: result of retrosynthesis.
         solved: boolean value indicating if the molecule was solved.
     """
+    if visited is None:
+        visited = set()
+    logger = context_logger.get()
+    if depth >= max_depth:
+        logger.warning(f"Max depth {max_depth} reached for {molecule}")
+        return {'type': 'mol', 'smiles': molecule, 'is_chemical': True, 'in_stock': False, 'children': []}, False
+    if molecule in visited:
+        logger.warning(f"Cycle detected: {molecule} already processed")
+        return {'type': 'mol', 'smiles': molecule, 'is_chemical': True, 'in_stock': False, 'children': []}, False
+    visited.add(molecule)
```

Comment:
@redrodeo03  is molecules added to set canonical? if not, duplicates may go undetected?